cpp/arcticdb/version/test/test_version_store.cpp

```diff
@@ -118,6 +118,27 @@ TEST(PythonVersionStore, DeleteAllVersions) {
     ASSERT_EQ(mock_store->num_atom_keys_of_type(KeyType::TABLE_INDEX), 0);
 }
 
+TEST(PythonVersionStore, WriteBadStreamId) {
+    using namespace arcticdb;
+    using namespace arcticdb::storage;
+    using namespace arcticdb::stream;
+    using namespace arcticdb::pipelines;
+
+    auto [version_store, mock_store] = python_version_store_in_memory();
+
+    std::string stream_id(1, 0);
```

Comment:
I find 

`stream_id{"."}`. Clearer.

I don't think we should validate against ".". We should permit at least all the "safe" characters in https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html and the "Characters that might require special handling" (Man uses some of them widely).

Reply:
This test is for "\0', not ".". But I agree, I will change it to stream_id{"\0"}.

Also all of the "safe" characters have been added to a test to make sure that they are supported.
The test is test_special_chars in test_basic_version_store.py

Reply:
But we also need the "Characters that might require special handling" to work